server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -213,9 +213,7 @@ public String toString() {
 
     private void onNoLongerPrimary(Exception failure) {
         final Throwable cause = ExceptionsHelper.unwrapCause(failure);
-        final boolean nodeIsClosing =
-                cause instanceof NodeClosedException
-                        || ExceptionsHelper.isTransportStoppedForAction(cause, "internal:cluster/shard/failure");
+        final boolean nodeIsClosing = cause instanceof NodeClosedException;
```

Comment:
I thought about this too, but ended up keeping this as is, since the unwrap's are different in that `unwrapCause` only unwraps `ElasticsearchWrapperException`, whereas `unwrap` unwraps all causes. It could be important in case we end up unwrapping a deep exception from another host? Do you think we should try to follow your suggestion anyway?

Reply:
Nah let's not do that. I def. doesn't look impossible for that exception from another node making it here (since it's not trivial to tell it the suggestion is pointless in the first place :)).